topic-operator/src/main/java/io/strimzi/operator/topic/v2/ReplicasChangeClient.java

```diff
@@ -0,0 +1,373 @@
+                ca = cf.generateCertificate(caInput);
+            } catch (IOException ioe) {
+                throw new RuntimeException(format("File not found: %s", config.cruiseControlCrtFilePath()));
+            }
+            // create a P12 keystore containing our trusted chain
+            KeyStore keyStore = KeyStore.getInstance("PKCS12");
+            keyStore.load(null, null);
+            keyStore.setCertificateEntry("ca", ca);
+            // create a trust manager that trusts the chain in our keystore
+            TrustManagerFactory tmf = TrustManagerFactory.getInstance("PKIX");
+            tmf.init(keyStore);
+            // create an SSL context that uses our trust manager
+            SSLContext sslContext = SSLContext.getInstance("TLS");
+            sslContext.init(null, tmf.getTrustManagers(), null);
+            builder.sslContext(sslContext);
+        }
+        return builder.build();
+    }
+
+    private HttpRequest buildTopicConfigPostRequest(List<ReconcilableTopic> reconcilableTopics) {
+        StringBuilder url = new StringBuilder(
+            format("%s://%s:%d%s?", config.cruiseControlSslEnabled() ? "https" : "http",
+                config.cruiseControlHostname(), config.cruiseControlPort(), CruiseControlEndpoints.TOPIC_CONFIGURATION));
+        url.append(format("%s=%s&", CruiseControlParameters.SKIP_RACK_AWARENESS_CHECK, !config.cruiseControlRackEnabled()));
+        url.append(format("%s=%s&", CruiseControlParameters.DRY_RUN, "false"));
+        url.append(format("%s=%s", CruiseControlParameters.JSON, "true"));
+        
+        Map<Integer, List<ReconcilableTopic>> topicsByReplicas = reconcilableTopics.stream()
+            .collect(groupingBy(rt -> rt.kt().getSpec().getReplicas()));
+        Map<Integer, String> requestPayload = new HashMap<>();
+        topicsByReplicas.entrySet().forEach(es -> {
+            int rf = es.getKey();
+            List<String> targetNames = topicNames(topicsByReplicas.get(rf));
+            requestPayload.put(rf, String.join("|", targetNames));
+        });
+        String json;
+        try {
+            json = mapper.writeValueAsString(
+                new ReplicationFactorChanges(new ReplicationFactor(requestPayload)));
+        } catch (JsonProcessingException e) {
+            throw new RuntimeException("Failed to serialize request body");
+        }
+        
+        LOGGER.traceOp("Request URL: {}, body: {}", url, json);
+        HttpRequest.Builder builder = HttpRequest.newBuilder()
+            .uri(URI.create(url.toString()))
+            .timeout(Duration.of(REQUEST_TIMEOUT_SEC, ChronoUnit.SECONDS))
+            .header("Content-Type", "application/json")
+            .POST(HttpRequest.BodyPublishers.ofString(json));
+        maybeAddBasicAuthHeader(builder);
+        return builder.build();
+    }
+    
+    private HttpRequest buildUserTasksGetRequest(Set<String> userTaskIds) {
```

Comment:
I know it's a `GET` request, but I think it's more natural to have the request method first, so `buildGetUserTasksRequest` and likewise `buildPostTopicConfigRequest`